### PR TITLE
Enable remote workflow to be invoked in conditional branch

### PR DESCRIPTION
**Pull request by ysysys3074**
# TL;DR
When invoking `reference_launch_plan` in flytekit conditional branch, it will fails at compile time with KeyError(root reason is a variable storing all the launch_plan does not inspect branch so it's empty and no key), this PR fix this issue to inspect all the launch_plan from all the branches including nested branches.

Fixes https://github.com/flyteorg/flyte/issues/4184
## Type
 - [x] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] E2E test done
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

unit test added and all passed:
Screenshot 2023-10-12 at 10 06 31 AM

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by pingsutw — COMMENTED** [review]
Thanks, LGTM, could you resolve the merge conflict?

**Review by pingsutw — DISMISSED** [review]

**Review by pingsutw — APPROVED** [review]
